lib/flex.js

```diff
@@ -0,0 +1,132 @@
+const Configuration = require('./configuration/configuration')
+const extractValidationStats = require('./stats/extractValidationStats')
+
+const SUPPORTED_VALIDATOR = [
+  'title',
+  'label',
+  'description',
+  'assignee',
+  'dependent',
+  'milestone',
+  'approval'
+]
+
+const SUPPORTED_ACTIONS = [
+  'comments',
+  'checks'
+]
+
+// Main logic Processor of mergeable
+const executeMergeable = async (context, registry) => {
+  if (registry === undefined) {
+    registry = { validators: new Map(), actions: new Map() }
+  }
+
+  // first fetch the configuration
+  let config = await Configuration.instanceWithContext(context)
+
+  // check config file and get settings only for the current event. the rest is useless for executor
+  let eventName = `${context.event}.${context.payload.action}`
+  config = filterByEvent(config, eventName)
+  // go through the settings and register all the validators
+  registerValidatorsAndActions(config, registry)
+
+  // do pre validation actions
+  await processPreActions(context, registry, config)
+  config.forEach(rule => {
+    Promise.all(getValiationPromises(context, registry, rule))
+      .then((result) => {
+        const translatedOutput = extractValidationStats(result)
+
+        Promise.all(getActionPromises(context, registry, rule, translatedOutput))
+      })
+  })
+}
+
+const filterByEvent = (config, eventName) => {
+  let filteredConfig = []
+  config.settings.mergeable.forEach(rule => {
+    let events = Array.isArray(rule.when) ? rule.when : rule.when.split(',').map(n => n.trim())
+    if (isEventInContext(eventName, events)) {
+      filteredConfig.push(rule)
+    }
+  })
+
+  return filteredConfig
+}
+
+const isEventInContext = (eventName, events) => {
```

Comment:
This function is not used anywhere. What is it for?

Reply:
it is being used in `filterByEvent` but i agree that configuration should be handling this code

Reply:
done